lib/utils/dotsnyk/index.ts

```diff
@@ -0,0 +1,46 @@
+import * as fs from 'fs';
+import { parse } from 'yaml';
+import { resolve } from 'path';
+
+// # Snyk (https://snyk.io) policy file
+// version: v1.14.0
+//
+// exclude:
+//   unmanaged:
+//     - test.spec.ts # single file
+//     - src/lib # single directory
+//     - tests/*.ts # any file with the “.ts” extension under “tests”
+//     - “**/*.spec.ts” # files ending with “.spec.ts” from any directory
+//     - tests?/* # files directly inside “test” and/or “tests” directories
+//     - tests/** # all files and directories inside “tests”
+//
+
+export type Config = { exclude: { unmanaged: string[] } };
+
+export function parseDotSnyk(filePath: string): Config {
+  try {
+    const dotSnykFile = fs.readFileSync(filePath, 'utf-8');
+    return parse(dotSnykFile);
+  } catch (err) {
+    if (err.code === 'EACCES' || err.code === 'EPERM') {
+      throw `${filePath} is not accessible.`;
+    }
+
+    if (err.code === 'ENOENT') {
+      throw `no such file or directory: ${filePath}`;
+    }
+
+    throw err;
+  }
+}
+
+export function parseUnmanagedExcludes(base: string, config: Config): string[] {
+  return toAbsolutePaths(base, config.exclude.unmanaged);
+}
+
+export function toAbsolutePaths(
```

Comment:
For completeness + peace of mind maybe we should test this too?

Reply:
Yep, this is copy-pasted from another file (used by scan.ts), and we use a copy to be able to extract dotsnyk/index.ts later to a separate lib.

So we could copy following test more or less: https://github.com/snyk/snyk-cpp-plugin/blob/master/test/scan.test.ts#L59